### ISSUE TITLE
QI-FLRC-1 Ceiling Fan, Light Controller: Connection to device succeeded but no datapoints found

### ISSUE THREAD

**Issue by dmperl** [rospogrigio/localtuya#1752]
## The problem

Fan configured with Tuya Smartlife.  Local Tuya configuration falis as follows:
![20240725 Tuya Fan new device config](https://github.com/user-attachments/assets/7b2a5a84-596c-46bd-b75e-310f709db6b5)

Error message:
![image](https://github.com/user-attachments/assets/d1080398-c75e-457f-982d-6776afd5d008)

## Environment

- Localtuya version:  5.2.1
- Home Assistant Core version:  core-2024.7.2
- [x] Does the device work using the Home Assistant Tuya Cloud component ? 
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [] Was the device working with earlier versions of localtuya ? Which one? 
- [x] Are you using the Tuya/SmartLife App in parallel ? 

## Steps to reproduce

1. Confgure with Smartlife App and Tuya (cloud) Home Assistant Integration  (both work)
2. Configure/Add device within Local Tuya (as shown above)
3. Error message displayed as shown above

It looks like other are having a similar issue with other devices.

## Provide Home Assistant traceback/logs

```
This error originated from a custom integration.

Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:863
integration: LocalTuya (documentation, issues)
First occurred: 1:19:47 PM (3 occurrences)
Last logged: 1:42:54 PM

[eba...pxs] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError

```

## Additional information

I have a working Temp&  Humidity device configured in LocalTuya.  It is a "read" type device.  The Fan Controller is a "controllable" type device.

Query Device Details works on the tuya IOT Platform.
      "product_id": "fcldsaadc3hzf9ey",
      "product_name": "吊扇灯",

tuya debugging -> Standard Instruction Set

![image](https://github.com/user-attachments/assets/60368102-adb9-4074-94c6-3bb1b486d848)
![image](https://github.com/user-attachments/assets/8cc10a76-80b2-468f-9c14-4cb9e84bb1cd)

**Comment by dmperl** [comment]
In the light of a new day I tried adding:

![image](https://github.com/user-attachments/assets/6562339d-658e-4eba-8dbd-ab6408b15b38)

in the form as:  (from the JSON EDITOR info above)

![image](https://github.com/user-attachments/assets/6ece9059-cfb1-486c-b459-75a9e411a989)


this got me a step further:

![image](https://github.com/user-attachments/assets/c9034a6b-16a7-4cba-837d-efecc96aa073)

Press Submit..  more "guesses" to make:

![image](https://github.com/user-attachments/assets/238120d9-fd05-45b8-a013-ece09909c625)
![image](https://github.com/user-attachments/assets/086e102a-3266-4408-a385-68c7cdae0dc1)

forgot to enter Friendly name*  ...   do over:

![image](https://github.com/user-attachments/assets/d570a594-c935-46d6-82e8-4358d680566a)

get **Unknown error occurred** message.   here are the new error-log entries:

This error originated from a custom integration.

Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:863
integration: LocalTuya (documentation, issues)
First occurred: 10:57:39 AM (1 occurrences)
Last logged: 10:57:39 AM

[eba...pxs] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError


Logger: aiohttp.server
Source: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:421
First occurred: 11:00:59 AM (1 occurrences)
Last logged: 11:00:59 AM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 210, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 285, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 74, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 122, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 368, in async_configure
    result = await self._async_configure(flow_id, user_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 415, in _async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 518, in _async_handle_step
    result: _FlowResultT = await getattr(flow, method)(user_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/config_flow.py", line 764, in async_step_configure_entity
    self.entities.append(strip_dps_values(user_input, self.dps_strings))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/config_flow.py", line 208, in strip_dps_values
    stripped[field] = int(user_input[field].split(" ")[0])
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'switch'

-----------------------------------------

**Comment by dmperl** [comment]
All the answers I was looking for are here:  https://github.com/dulfer/localtuya-device-datapoints

Switch = 1
Voice = 17
Wind Speed Level = 3
Countdown Timer = 6
Remaining Time = 7
Light = 9

**Comment by dmperl** [comment]
Next Question:  After I enter all of the DP ids for this device than "values" all show as -1.    How do I associate the proper values with each Data Point type?
![image](https://github.com/user-attachments/assets/6cfe85db-df65-4f62-acad-70e5254f03df)

**Comment by dmperl** [comment]
Initial problem solved.  Closing this issue.
Will open new issue for "next" problem
